src/main/java/com/fasterxml/jackson/databind/deser/std/StringDeserializer.java

```diff
@@ -61,6 +62,15 @@ public String deserialize(JsonParser p, DeserializationContext ctxt) throws IOEx
         if (t == JsonToken.START_OBJECT) {
             return ctxt.extractScalarFromObject(p, this, _valueClass);
         }
+        if (t == JsonToken.VALUE_NUMBER_INT) {
```

Comment:
Could this be simplified by calling `_parseString(...)` and removing handling of START_OBJECT, VALUE_EMBEDDED_OBJECT (that are handled there)?

Reply:
Good one! Yes, it can. I've rewritten this method in terms of `_parseString(...)`, leaving only the `START_ARRAY` case in the original method and the `VALUE_STRING` case at the very top as an optimization — I assume it is the most common case, so I figured it would make sense to keep that there even if it is already handled in `_parseString(...)`. If you think this is a premature optimization, let me know and I'll remove it :-)